### ISSUE TITLE
Type of TypedDict field "property" cannot be redefined ​Pyright

### ISSUE THREAD

**Issue by predragnikolic** [microsoft/pyright#3791]
Hello  :wave: 

**Describe the bug**

When type `B` extends `A`,
if `B` has the same property name as `A`
and the property type of `B` (`Literal['some_string']`) 
is compatible with property type of `A` (`str`),
I expect no errors to be reported by the type checker.

**To Reproduce**
Given the following code:
```py
from typing import Literal, TypedDict

class A(TypedDict):
	kind: str

class B(A, TypedDict):
	kind: Literal['create']
```

**Expected behavior**
I expect to have no errors.

**Current behavior**
But pyright reports me an error:
```py
class B(A, TypedDict):
	kind: Literal['create']
      # ^^^^ error - Type of TypedDict field "kind" cannot be redefined ​Pyright:
      #              Type in parent class is "str" and type in child class is "Literal['create']"

```

**Comment by erictraut** [comment]
Pyright is correct in generating an error here. It is not type safe to redefine the type of a key in `TypedDict` derived from another `TypedDict`. Consider the following:

```python
def update_kind(td: A):
    td["kind"] = "not_create"

td: B = {"kind": "create"}
update_kind(td)
```

Mypy also generates an error in this case.
